### ISSUE TITLE
Update list of files to remove after migration

### ISSUE THREAD

**Issue by nylen** [ClassicPress/ClassicPress#433]
There are now more WordPress 5.x files that we should remove after migrating to ClassicPress.  This change is done in the ClassicPress core code, and we can include it in the upcoming v1.1.0.

We can use the Shield Security plugin (or other methods such as installing a site within a `git` repository) to get a list of the files to be removed.

See: https://forums.classicpress.net/t/identifying-redundant-js-files/1245

Previously (use this change as a guideline): #286

**Comment by ElisabettaCarrara** [comment]
[wp-cp-diff.zip](https://github.com/ClassicPress/ClassicPress/files/3400759/wp-cp-diff.zip)
I used meld software to obtain the comparison among a complete wp folder and a complete cp folder.
Since meld hasn't the export to file feature. I was left with the option to screenshot it.

**Comment by Norskes** [comment]
```
<?php
/**
 * Redunadant files from WP 5.2.2
 * (Clean install + migration. Folder 'wp-content' excluded)
 */

$redundant_files = array(

	// JS includes

	'wp-includes\js\backbone.js',
	'wp-includes\js\clipboard.js',
	'wp-includes\js\clipboard.min.js',
	'wp-includes\js\codemirror\esprima.js',
	'wp-includes\js\codemirror\fakejshint.js',
	'wp-includes\js\tinymce\wp-tinymce.js',
	'wp-includes\js\underscore.js',


	// Site health feature "looking up a site's health based on a user's WordPress environment. @since 5.2.0"

	'wp-admin\css\site-health.css',
	'wp-admin\css\site-health-rtl.css',
	'wp-admin\css\site-health-rtl.min.css',
	'wp-admin\css\site-health.min.css',
	'wp-admin\js\site-health.js',
	'wp-admin\js\site-health.min.js',
	'wp-admin\site-health.php',
	'wp-admin\site-health-info.php',
	'wp-admin\includes\class-wp-site-health.php',
	'wp-admin\includes\class-wp-site-health-auto-updates.php',


	// "Class for providing debug data based on a users WordPress environment. @since 5.2.0"

	'wp-admin\includes\class-wp-debug-data.php',


	// "Error Protection API (@since   5.2.0)

	'wp-includes\class-wp-fatal-error-handler.php',
	'wp-includes\class-wp-paused-extensions-storage.php',
	'wp-includes\class-wp-recovery-mode.php',
	'wp-includes\class-wp-recovery-mode-cookie-service.php',
	'wp-includes\class-wp-recovery-mode-email-service.php',
	'wp-includes\class-wp-recovery-mode-key-service.php',
	'wp-includes\class-wp-recovery-mode-link-service.php',
	'wp-includes\error-protection.php',


	// Network API (@since 5.1.0)

	'wp-includes\ms-network.php',


	// Site API (@since 5.1.0)

	'wp-includes\ms-site.php',


	// "Sodium Compat is a pure PHP polyfill for the Sodium cryptography library (libsodium), a core extension in PHP 7.2.0 ..."

	'wp-includes\sodium_compat\composer.json',
	'wp-includes\sodium_compat\src\Core32\Curve25519\README.md',
	'wp-includes\sodium_compat\src\Core\Curve25519\README.md',
	'wp-includes\sodium_compat\autoload.php',
	'wp-includes\sodium_compat\lib\constants.php',
	'wp-includes\sodium_compat\lib\namespaced.php',
	'wp-includes\sodium_compat\lib\php72compat.php',
	'wp-includes\sodium_compat\lib\sodium_compat.php',
	'wp-includes\sodium_compat\namespaced\Compat.php',
	'wp-includes\sodium_compat\namespaced\Core\BLAKE2b.php',
	'wp-includes\sodium_compat\namespaced\Core\ChaCha20.php',
	'wp-includes\sodium_compat\namespaced\Core\ChaCha20\Ctx.php',
	'wp-includes\sodium_compat\namespaced\Core\ChaCha20\IetfCtx.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Fe.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Ge\Cached.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Ge\P1p1.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Ge\P2.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Ge\P3.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\Ge\Precomp.php',
	'wp-includes\sodium_compat\namespaced\Core\Curve25519\H.php',
	'wp-includes\sodium_compat\namespaced\Core\Ed25519.php',
	'wp-includes\sodium_compat\namespaced\Core\HChaCha20.php',
	'wp-includes\sodium_compat\namespaced\Core\HSalsa20.php',
	'wp-includes\sodium_compat\namespaced\Core\Poly1305.php',
	'wp-includes\sodium_compat\namespaced\Core\Poly1305\State.php',
	'wp-includes\sodium_compat\namespaced\Core\Salsa20.php',
	'wp-includes\sodium_compat\namespaced\Core\SipHash.php',
	'wp-includes\sodium_compat\namespaced\Core\Util.php',
	'wp-includes\sodium_compat\namespaced\Core\X25519.php',
	'wp-includes\sodium_compat\namespaced\Core\XChaCha20.php',
	'wp-includes\sodium_compat\namespaced\Core\Xsalsa20.php',
	'wp-includes\sodium_compat\namespaced\Crypto.php',
	'wp-includes\sodium_compat\namespaced\File.php',
	'wp-includes\sodium_compat\src\Compat.php',
	'wp-includes\sodium_compat\src\Core32\BLAKE2b.php',
	'wp-includes\sodium_compat\src\Core32\ChaCha20.php',
	'wp-includes\sodium_compat\src\Core32\ChaCha20\Ctx.php',
	'wp-includes\sodium_compat\src\Core32\ChaCha20\IetfCtx.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Fe.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Ge\Cached.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Ge\P1p1.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Ge\P2.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Ge\P3.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\Ge\Precomp.php',
	'wp-includes\sodium_compat\src\Core32\Curve25519\H.php',
	'wp-includes\sodium_compat\src\Core32\Ed25519.php',
	'wp-includes\sodium_compat\src\Core32\HChaCha20.php',
	'wp-includes\sodium_compat\src\Core32\HSalsa20.php',
	'wp-includes\sodium_compat\src\Core32\Int32.php',
	'wp-includes\sodium_compat\src\Core32\Int64.php',
	'wp-includes\sodium_compat\src\Core32\Poly1305.php',
	'wp-includes\sodium_compat\src\Core32\Poly1305\State.php',
	'wp-includes\sodium_compat\src\Core32\Salsa20.php',
	'wp-includes\sodium_compat\src\Core32\SipHash.php',
	'wp-includes\sodium_compat\src\Core32\Util.php',
	'wp-includes\sodium_compat\src\Core32\X25519.php',
	'wp-includes\sodium_compat\src\Core32\XChaCha20.php',
	'wp-includes\sodium_compat\src\Core32\XSalsa20.php',
	'wp-includes\sodium_compat\src\Core\BLAKE2b.php',
	'wp-includes\sodium_compat\src\Core\ChaCha20.php',
	'wp-includes\sodium_compat\src\Core\ChaCha20\Ctx.php',
	'wp-includes\sodium_compat\src\Core\ChaCha20\IetfCtx.php',
	'wp-includes\sodium_compat\src\Core\Curve25519.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Fe.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Ge\Cached.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Ge\P1p1.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Ge\P2.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Ge\P3.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\Ge\Precomp.php',
	'wp-includes\sodium_compat\src\Core\Curve25519\H.php',
	'wp-includes\sodium_compat\src\Core\Ed25519.php',
	'wp-includes\sodium_compat\src\Core\HChaCha20.php',
	'wp-includes\sodium_compat\src\Core\HSalsa20.php',
	'wp-includes\sodium_compat\src\Core\Poly1305.php',
	'wp-includes\sodium_compat\src\Core\Poly1305\State.php',
	'wp-includes\sodium_compat\src\Core\Salsa20.php',
	'wp-includes\sodium_compat\src\Core\SipHash.php',
	'wp-includes\sodium_compat\src\Core\Util.php',
	'wp-includes\sodium_compat\src\Core\X25519.php',
	'wp-includes\sodium_compat\src\Core\XChaCha20.php',
	'wp-includes\sodium_compat\src\Core\XSalsa20.php',
	'wp-includes\sodium_compat\src\Crypto.php',
	'wp-includes\sodium_compat\src\Crypto32.php',
	'wp-includes\sodium_compat\src\File.php',
	'wp-includes\sodium_compat\src\SodiumException.php',


	// Woff2 format for dashicons (should be generated on build, probably)

	'wp-includes\fonts\dashicons.woff2',
);
```

**Comment by nylen** [comment]
Reminder to change `\` backslashes to `/` forward slashes. It would also be a good idea to test this to make sure there are no other files in newer versions, and then it should be good to go.

**Comment by Norskes** [comment]
Compared once more (clean installation of the latest WP 5.2.3, then migration by a plugin; suppressed  WP max version by a filter). Used a gulp task based on fs.statSync to check and compare mtime attributes). Same result:
```
<?php
	'wp-admin/css/site-health-rtl.css',
	'wp-admin/css/site-health-rtl.min.css',
	'wp-admin/css/site-health.css',
	'wp-admin/css/site-health.min.css',
	'wp-admin/includes/class-wp-debug-data.php',
	'wp-admin/includes/class-wp-site-health-auto-updates.php',
	'wp-admin/includes/class-wp-site-health.php',
	'wp-admin/js/site-health.js',
	'wp-admin/js/site-health.min.js',
	'wp-admin/site-health-info.php',
	'wp-admin/site-health.php',
	'wp-includes/class-wp-fatal-error-handler.php',
	'wp-includes/class-wp-paused-extensions-storage.php',
	'wp-includes/class-wp-recovery-mode-cookie-service.php',
	'wp-includes/class-wp-recovery-mode-email-service.php',
	'wp-includes/class-wp-recovery-mode-key-service.php',
	'wp-includes/class-wp-recovery-mode-link-service.php',
	'wp-includes/class-wp-recovery-mode.php',
	'wp-includes/error-protection.php',
	'wp-includes/fonts/dashicons.woff2',
	'wp-includes/js/backbone.js',
	'wp-includes/js/clipboard.js',
	'wp-includes/js/clipboard.min.js',
	'wp-includes/js/codemirror/esprima.js',
	'wp-includes/js/codemirror/fakejshint.js',
	'wp-includes/js/tinymce/wp-tinymce.js',
	'wp-includes/js/underscore.js',
	'wp-includes/ms-network.php',
	'wp-includes/ms-site.php',
	'wp-includes/sodium_compat/autoload.php',
	'wp-includes/sodium_compat/composer.json',
	'wp-includes/sodium_compat/lib/constants.php',
	'wp-includes/sodium_compat/lib/namespaced.php',
	'wp-includes/sodium_compat/lib/php72compat.php',
	'wp-includes/sodium_compat/lib/sodium_compat.php',
	'wp-includes/sodium_compat/namespaced/Compat.php',
	'wp-includes/sodium_compat/namespaced/Core/BLAKE2b.php',
	'wp-includes/sodium_compat/namespaced/Core/ChaCha20.php',
	'wp-includes/sodium_compat/namespaced/Core/ChaCha20/Ctx.php',
	'wp-includes/sodium_compat/namespaced/Core/ChaCha20/IetfCtx.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Fe.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Ge/Cached.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Ge/P1p1.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Ge/P2.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Ge/P3.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/Ge/Precomp.php',
	'wp-includes/sodium_compat/namespaced/Core/Curve25519/H.php',
	'wp-includes/sodium_compat/namespaced/Core/Ed25519.php',
	'wp-includes/sodium_compat/namespaced/Core/HChaCha20.php',
	'wp-includes/sodium_compat/namespaced/Core/HSalsa20.php',
	'wp-includes/sodium_compat/namespaced/Core/Poly1305.php',
	'wp-includes/sodium_compat/namespaced/Core/Poly1305/State.php',
	'wp-includes/sodium_compat/namespaced/Core/Salsa20.php',
	'wp-includes/sodium_compat/namespaced/Core/SipHash.php',
	'wp-includes/sodium_compat/namespaced/Core/Util.php',
	'wp-includes/sodium_compat/namespaced/Core/X25519.php',
	'wp-includes/sodium_compat/namespaced/Core/XChaCha20.php',
	'wp-includes/sodium_compat/namespaced/Core/Xsalsa20.php',
	'wp-includes/sodium_compat/namespaced/Crypto.php',
	'wp-includes/sodium_compat/namespaced/File.php',
	'wp-includes/sodium_compat/src/Compat.php',
	'wp-includes/sodium_compat/src/Core32/BLAKE2b.php',
	'wp-includes/sodium_compat/src/Core32/ChaCha20.php',
	'wp-includes/sodium_compat/src/Core32/ChaCha20/Ctx.php',
	'wp-includes/sodium_compat/src/Core32/ChaCha20/IetfCtx.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Fe.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Ge/Cached.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Ge/P1p1.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Ge/P2.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Ge/P3.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/Ge/Precomp.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/H.php',
	'wp-includes/sodium_compat/src/Core32/Curve25519/README.md',
	'wp-includes/sodium_compat/src/Core32/Ed25519.php',
	'wp-includes/sodium_compat/src/Core32/HChaCha20.php',
	'wp-includes/sodium_compat/src/Core32/HSalsa20.php',
	'wp-includes/sodium_compat/src/Core32/Int32.php',
	'wp-includes/sodium_compat/src/Core32/Int64.php',
	'wp-includes/sodium_compat/src/Core32/Poly1305.php',
	'wp-includes/sodium_compat/src/Core32/Poly1305/State.php',
	'wp-includes/sodium_compat/src/Core32/Salsa20.php',
	'wp-includes/sodium_compat/src/Core32/SipHash.php',
	'wp-includes/sodium_compat/src/Core32/Util.php',
	'wp-includes/sodium_compat/src/Core32/X25519.php',
	'wp-includes/sodium_compat/src/Core32/XChaCha20.php',
	'wp-includes/sodium_compat/src/Core32/XSalsa20.php',
	'wp-includes/sodium_compat/src/Core/BLAKE2b.php',
	'wp-includes/sodium_compat/src/Core/ChaCha20.php',
	'wp-includes/sodium_compat/src/Core/ChaCha20/Ctx.php',
	'wp-includes/sodium_compat/src/Core/ChaCha20/IetfCtx.php',
	'wp-includes/sodium_compat/src/Core/Curve25519.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Fe.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Ge/Cached.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Ge/P1p1.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Ge/P2.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Ge/P3.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/Ge/Precomp.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/H.php',
	'wp-includes/sodium_compat/src/Core/Curve25519/README.md',
	'wp-includes/sodium_compat/src/Core/Ed25519.php',
	'wp-includes/sodium_compat/src/Core/HChaCha20.php',
	'wp-includes/sodium_compat/src/Core/HSalsa20.php',
	'wp-includes/sodium_compat/src/Core/Poly1305.php',
	'wp-includes/sodium_compat/src/Core/Poly1305/State.php',
	'wp-includes/sodium_compat/src/Core/Salsa20.php',
	'wp-includes/sodium_compat/src/Core/SipHash.php',
	'wp-includes/sodium_compat/src/Core/Util.php',
	'wp-includes/sodium_compat/src/Core/X25519.php',
	'wp-includes/sodium_compat/src/Core/XChaCha20.php',
	'wp-includes/sodium_compat/src/Core/XSalsa20.php',
	'wp-includes/sodium_compat/src/Crypto.php',
	'wp-includes/sodium_compat/src/Crypto32.php',
	'wp-includes/sodium_compat/src/File.php',
	'wp-includes/sodium_compat/src/SodiumException.php',
?>
```

Not sure what actions should be performed to test this deeper. Please, let me know if I can do something else for this issue.

**Comment by nylen** [comment]
Thanks @Norskes, I've put up a PR to close this one out: #487

Here is what I did, by the way, documenting it for next time. If you think your `grunt` task may be easier then feel free to share, as this task will come up again :)

- install WP 5.2.3
- put the site in a `git` repository and `git commit`
- migrate to ClassicPress
- `git commit` again

```
git remote add nightly https://github.com/ClassyBot/ClassicPress-nightly
git fetch nightly
diff -u <(git ls-files) <(git ls-tree -r --name-only 1.0.2+migration.20190905 | sed 's#^wordpress/##') | grep '^-[^-]' | cut -c2- > to-delete.txt
```

This gets a list of files that are present in this folder but not in the migration build of ClassicPress. It includes some things that aren't relevant here like `wp-config.php` and plugins and themes, but other than that it is the same as your list.

Then in order to get to the PR we need to figure out where entire directories can be deleted, `wp-includes/sodium_compat/` is the only one this time around.

---

>migration by a plugin; suppressed WP max version by a filter

I've also updated the migration system to support WP 5.2.3 without having to use that filter: https://github.com/ClassicPress/ClassicPress-APIs/compare/f815502...3d1c830
